internal/services/mssql/mssql_elasticpool_resource.go

```diff
@@ -347,6 +355,12 @@ func resourceMsSqlElasticPoolRead(d *pluginsdk.ResourceData, meta interface{}) e
 			}
 			d.Set("license_type", licenseType)
 
+			highAvailabilityReplicaCount := int64(1)
+			if props.HighAvailabilityReplicaCount != nil {
+				highAvailabilityReplicaCount = pointer.From(props.HighAvailabilityReplicaCount)
+			}
+			d.Set("high_availability_replica_count", pointer.To(highAvailabilityReplicaCount))
```

Comment:
The read func should set what was returned by Azure here, if Azure didn't return a value for this, Terraform should set "null".

```suggestion
			d.Set("high_availability_replica_count", props.HighAvailabilityReplicaCount)
```